drivers/lora/sx12xx_common.c

```diff
@@ -153,6 +174,24 @@ static void sx12xx_ev_rx_done(uint8_t *payload, uint16_t size, int16_t rssi,
 	k_poll_signal_raise(sig, 0);
 }
 
+static void sx12xx_ev_cad_done(bool detected)
+{
+	if(detected) {
+		struct k_poll_signal *sig = dev_data.operation_done;
+
+		if (modem_release(&dev_data)) {
+			/* Raise signal if provided */
+			if (sig) {
```

Comment:
Silently not sending data if CAD fails without a signal being provided does not seem like a great idea.
I would propose enforcing back at the `lora_send_async` function that a signal is provided if CAD is enabled